plugins/pur/core/src/main/java/org/pentaho/di/repository/pur/PurRepository.java

```diff
@@ -1890,7 +2050,14 @@ private ObjectId renameTransOrJob( ObjectId idObject, String versionComment,
                                      RepositoryDirectoryInterface newDirectory, String newTitle,
                                      RepositoryObjectType objectType, String errorMsgKey ) throws KettleException {
 
-    RepositoryFile file = pur.getFileById( idObject.getId() );
+    readWriteLock.readLock().lock();
+    RepositoryFile file;
+    try {
+      file = pur.getFileById( idObject.getId() );
+    } finally {
+      readWriteLock.readLock().unlock();
```

Comment:
The Write lock should not be release between this and lower in the same method. The lock should be kept.